docs/sphinx/examples/cpp/ionq

```diff
@@ -0,0 +1,52 @@
+// Compile and run with:
+// ```
+// nvq++ --target ionq ionq.cpp -o out.x && ./out.x
+// ```
+// Assumes a valid set of credentials have been stored. 
+
+#include <cudaq.h>
+#include <fstream>
+
+// Define a simple quantum kernel to execute on IonQ.
+struct ghz {
+  // Maximally entangled state between 5 qubits.
+  auto operator()() __qpu__ {
+    cudaq::qreg q(5);
+    h(q[0]);
+    for (int i = 0; i < 4; i++) {
+      x<cudaq::ctrl>(q[i], q[i + 1]);
+    }
+    mz(q);
+  }
+};
+
+int main() {
+  // Submit to IonQ synchronously. E.g, wait for the job
+  // result to be returned before proceeding. 
+  auto counts = cudaq::sample(ghz{});
+  counts.dump();
+
+  // Submit to IonQ asynchronously. E.g, continue executing
+  // code in the file until the job has been returned.
+  auto future = cudaq::sample_async(ghz{});
+  // ... classical code to execute in the meantime ...
+
+  // Can either grab the future result as soon as it's ready:
+  auto async_counts_now = future.get();
```

Comment:
It would make more sense to me to show that after reading it back in from the file - i.e write to file when results are not yet in, then retrieve it later (usually in a different program) and read/get the results.